src/rt/gtfsrt_update.cc

```diff
@@ -447,10 +455,132 @@ statistics gtfsrt_update_msg(timetable const& tt,
       }
     };
 
-    unsupported(entity.has_vehicle(), "vehicle", stats.unsupported_vehicle_);
     unsupported(entity.has_is_deleted() && entity.is_deleted(), "deleted",
                 stats.unsupported_deleted_);
 
+    if (entity.has_vehicle()) {
+      if (!entity.vehicle().has_position()) {
+        log(log_lvl::error, "rt.gtfs.unsupported",
+            R"(unsupported: no "position" field in "vehicle_position" field (tag={}, id={}), skipping message)",
+            tag, entity.id());
+        ++stats.vehicle_position_without_position_;
+      } else if (!entity.vehicle().has_trip()) {
+        log(log_lvl::error, "rt.gtfs.unsupported",
+            R"(unsupported: no "trip" field in "vehicle_position" field (tag={}, id={}), skipping message)",
+            tag, entity.id());
+        ++stats.vehicle_position_without_trip_;
+      } else if (!entity.vehicle().trip().has_trip_id()) {
+        log(log_lvl::error, "rt.gtfs.unsupported",
+            R"(unsupported: no "trip_id" field in "trip" field (tag={}, id={}), skipping message)",
+            tag, entity.id());
+        ++stats.vehicle_position_trip_without_trip_id_;
+      } else if (!entity.vehicle().trip().has_route_id()) {
+        log(log_lvl::error, "rt.gtfs.unsupported",
+            R"(unsupported: no "route_id" field in "trip" field (tag={}, id={}), skipping message)",
+            tag, entity.id());
+        ++stats.vehicle_position_trip_without_route_id_;
+      } else {
+        auto const vp = entity.vehicle();
```

Comment:
separate function (like `update_run`)

should also only be done if a flag from outside says to use `VehiclePosition` updates as most feeds that provide them also have `TripUpdates` which makes `VehiclePosition` delay predictions obsolete